README.md

```diff
@@ -21,39 +21,238 @@ See the [Curity OAuth for Web Home Page](https://curity.io/product/token-service
 
 The plugin can be used standalone, or in conjunction with the [Phantom Token Plugin](https://curity.io/resources/learn/phantom-token-pattern/):
 
-![API Flow](/doc/api-flow.png)
+![API Flow](/images/api-flow.png)
 
 See also the following resources:
 
 - The [Example SPA](https://github.com/curityio/web-oauth-via-bff), which acts as a client to this plugin.
-- The [OAuth Agent API](https://github.com/curityio/bff-node-express), which issues the secure cookies for the SPA.
+- The [OAuth Agent API](https://github.com/curityio/token-handler-node-express), which issues the secure cookies for the SPA.
 
-## Configuration
+## Required Configuration Directives
 
-The plugin is configured with the following properties and decrypts AES256 encrypted cookies:
+All of the settings in this section are required:
 
-| Property | Description |
-| -------- | ----------- |
-| Encryption Key | The encryption key used by the plugin to decrypt AES256 encrypted SameSite cookies |
-| Cookie Name Prefix | The prefix used in the SPA's cookie name, typically representing a company or product |
-| Trusted Web Origins | The web origins from which the plugin will accept cookie requests |
+#### cookie_name_prefix
 
-## Defining Routes
+> **Syntax**: **`cookie_name_prefix`** `string`
+>
+> **Default**: *``*
+>
+> **Context**: `location`
 
-The plugin allows you to configure reverse proxy routes the same for web and mobile clients of your APIs:
+The prefix used in the SPA's cookie name, typically representing a company or product name.\
+The value supplied must not be empty, and `example` would lead to full cookie names such as `cookie-at`.
 
-| Client Type | Routing Behavior |
-| ----------- | ---------------- |
-| SPA | No authorization header is sent, and one is calculated from secure cookies received |
-| Mobile | If an authorization header is received, it is passed straight through to the API |
+#### encryption_key
 
-## Deployment and Testing
+> **Syntax**: **`encryption_key`** `string`
+>
+> **Default**: *``*
+>
+> **Context**: `location`
 
-The plugin can run in any NGINX based system with the LUA module enabled.
-See the [NGINX LUA OAuth Proxy Plugin](https://curity.io/resources/learn/oauth-proxy-plugin-lua) tutorial for further details.
+This must be a 32 byte encryption key expressed as 64 hex characters.\
+It is used to decrypt AES256 encrypted secure cookies.\
+The key is initially generated with a tool such as `openssl`, as explained in Curity tutorials.
 
-- [Kong Open Source](/doc/kong.md)
-- [OpenResty](/doc/openresty.md)
+#### trusted_web_origins
+
+> **Syntax**: **`trusted_web_origins`** `string[]`
+>
+> **Default**: *[]*
+>
+> **Context**: `location`
+
+A whitelist of at least one web origin from which the plugin will accept requests.\
+Multiple origins could be used in special cases where cookies are shared across subdomains.
+
+#### cors_enabled
+
+> **Syntax**: **`cors_enabled`** `boolean`
+>
+> **Default**: *true*
```

Comment:
I don't think a required option should have a default value. Maybe `cors_enabled` should be optional and indeed have a default value of `true`?

Reply:
Done